### ISSUE TITLE
Parsing logs into sqlite with default values is sloooooow

### ISSUE THREAD

**Issue by nphilipp** [fedora-infra/mirrors-countme#51]
# Story

As a user of DNF mirror statistics,
I want log files to be parsed quickly,
so that I don’t have to wait for results for too long.

As a planet,
I want that regularly performed automated tasks are performing efficiently,
so I don’t run out of breath.

# Acceptance Criteria

- [ ] Parsing 10GB of log data (roughly a data worth of it) and writing them into the raw sqlite database takes on the order of minutes, not hours.

# Background

The log file I use for testing contains roughly 3 million "countme" items and took 25 mins to process ~4%, while going ever slower (at which point I killed the script). Deduplicating currently looks like it does a linear search across the whole `countme_raw` table because there's no index across all the columns. This results in a total of ~4.5 trillion rows to be looked at when checking if all 3 million candidates are duplicates or not.

**Comment by james-antill** [comment]
There is an index on timestamp ... which should drop the linear check from millions to a couple of hundred at most.
Obviously just opening the sqlite DB in the cli and manually adding an index to see how performance changes should be easy to do/test ... but I wouldn't be shocked if something else is happening.

**Comment by nphilipp** [comment]
Yeah, I figured the existing index would make things a little less bad (if the SQLite query planner is up to snuff at least). I did that experiment, and with an index over all columns execution times are on par with CSV output, something around 4 minutes on my laptop.

**Comment by nphilipp** [comment]
So, the real reason for this is that the index is only created after all the data is written. I.e. for the first time, this will do linear searches to deduplicate because the index doesn’t exist yet.
